packages/network/test/integration/browser-ws-endpoint.test.ts

```diff
@@ -93,4 +93,23 @@ describe('ws-endpoint', () => {
             expect(close[1]).toContain('Handshake not received from connection behind UUID')
         })
     })
+
+    describe('Duplicate connections from same nodeId are closed', () => {
+        it('Duplicate connection is closed', async () => {
+            const client1 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+            const client2 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+
+            const server = await startServerWsEndpoint('127.0.0.1', 38482, PeerInfo.newNode('server'))
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
+                expect(err).toContain('private key')
```

Comment:
Maybe more fitting given the context of the test
```suggestion
                expect(err).toContain('Duplicate nodeId detected')
```